### PR TITLE
Implemented revert functionality#22xdxhp

### PR DESCRIPTION
**Pull request by disha1202**

### CODE REVIEW

File: src/components/ProductPopover.vue [hunk 1]
```diff
@@ -31,6 +32,9 @@ export default defineComponent({
     }),
   },
   methods: {
+    onlyRevert() {
+      this.isVirtual ? this.onlyRevertSingleProduct() : this.onlyRevertParentProduct();
```
Comment (adityasharma7) [inline src/components/ProductPopover.vue:36]: If isVirtual means it is virtual product, I think we should use onlyRevertParentProduct and for variant we should use onlyRevertSingleProduct

File: src/components/ProductPopover.vue [hunk 2]
```diff
@@ -31,6 +32,9 @@ export default defineComponent({
     }),
   },
   methods: {
+    onlyRevert() {
```
Comment (adityasharma7) [inline src/components/ProductPopover.vue:35]:
```suggestion
    revert() {
```

File: src/components/ProductPopover.vue [hunk 3]
```diff
@@ -45,12 +49,41 @@ export default defineComponent({
         element.isSelected = element.internalName === this.id;
       });
       popoverController.dismiss({ dismissed: true });
+    },
+    onlyRevertSingleProduct() {
```
Comment (adityasharma7) [inline src/components/ProductPopover.vue:53]:
```suggestion
    revertVariantProduct() {
```

File: src/components/ProductPopover.vue [hunk 4]
```diff
@@ -45,12 +49,41 @@ export default defineComponent({
...
+            return item.internalName === this.id;
+          })
+          element = item;
+        }
+        return element;
+      });
+      this.store.dispatch('order/updatedOrderListItems', items)
+      popoverController.dismiss({ dismissed: true });
+    },
+    onlyRevertParentProduct(){
```
Comment (adityasharma7) [inline src/components/ProductPopover.vue:67]:
```suggestion
    revertParentProduct(){
```

File: src/views/OrderDetail.vue [hunk 5]
```diff
@@ -329,6 +329,10 @@ export default defineComponent({
     selectProduct(item: any, event: any) {
       item.isSelected = event.detail.checked;
     },
+    revertAllChanges() {
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:332]:
```suggestion
    revertAll() {
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/ProductPopover.vue (hunk 1 above)
Comment (disha1202) [reply src/components/ProductPopover.vue]: Updated the code.

File: src/components/ProductPopover.vue (hunk 2 above)
Comment (disha1202) [reply src/components/ProductPopover.vue]: Made the requested change.

File: src/components/ProductPopover.vue (hunk 3 above)
Comment (disha1202) [reply src/components/ProductPopover.vue]: Made the requested change.

File: src/components/ProductPopover.vue (hunk 4 above)
Comment (disha1202) [reply src/components/ProductPopover.vue]: Made the requested change.

File: changelogs/unreleased/-22xdxhp.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+title: Implemented revert functionality(#22xdxhp)
```
Comment (ymaheshwari1) [inline changelogs/unreleased/-22xdxhp.yml:2]: Remove ticket id from changelog.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/-22xdxhp.yml (hunk 6 above)
Comment (disha1202) [reply changelogs/unreleased/-22xdxhp.yml]: Removed.

File: src/views/OrderDetail.vue (hunk 5 above)
Comment (disha1202) [reply src/views/OrderDetail.vue]: Made the requested change.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**